src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
+Animal.removeKilled = function() {
+  Animal.alive = Animal.alive.filter(animal => animal.health > 0);
+};
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim instanceof Herbivore && !victim.hidden) {
+      victim.health -= 50;
+      Animal.removeKilled();
```

Comment:
In my opinion it would be good to check here `victim.health` and if it is `<=0` then you could call the `Animal.removeKilled` function. There is no need to filter Animals when there is no need to. Imagine if we will have some more amout of data that might cost a lot of performance.

- [ ] Conditionaly call `Animal.removeKilled()`

Reply:
okay thanks. Good suggestion :)!